src/route/dataset-route.ts

```diff
@@ -174,13 +201,47 @@ apiRoute.get('/:dataset/xlsx', async (req, res) => {
     });
 });
 
+apiRoute.get('/:dataset/preview', async (req, res) => {
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
```

Comment:
There's another Response code for Bad Request.

Reply:
Changed response code to `400`